common/djangoapps/third_party_auth/management/commands/tests/test_saml.py

```diff
@@ -334,59 +329,104 @@ def test_xml_parse_exceptions(self, mocked_get):
             call_command("saml", pull=True, stdout=self.stdout)
         assert expected in self.stdout.getvalue()
 
-    @data(
-        (True, '[DRY RUN]', 'should not update provider configs'),
-        (False, '', 'should create new provider config for new version')
-    )
-    @unpack
-    def test_fix_references(self, dry_run, expected_output_marker, test_description):
+    def _run_checks_command(self):
+        """
+        Helper method to run the --run-checks command and return output.
         """
-        Test the --fix-references command with and without --dry-run option.
+        out = StringIO()
+        call_command('saml', '--run-checks', stdout=out)
+        return out.getvalue()
 
-        Args:
-            dry_run (bool): Whether to run with --dry-run flag
-            expected_output_marker (str): Expected marker in output
-            test_description (str): Description of what the test should do
+    @mock.patch('common.djangoapps.third_party_auth.management.commands.saml.set_custom_attribute')
+    def test_run_checks_outdated_configs(self, mock_set_custom_attribute):
+        """
+        Test the --run-checks command identifies outdated configurations.
         """
-        old_config, new_config, test_provider_config = self._setup_test_configs_for_fix_references()
-        new_config_id = new_config.id
-        original_config_id = old_config.id
+        old_config, new_config, test_provider_config = self._setup_test_configs_for_run_checks()
 
-        out = StringIO()
-        if dry_run:
-            call_command('saml', '--fix-references', '--dry-run', stdout=out)
-        else:
-            call_command('saml', '--fix-references', stdout=out)
+        output = self._run_checks_command()
+
+        self.assertIn('[WARNING]', output)
+        self.assertIn('test-provider', output)
+        self.assertIn(
+            f'id={old_config.id} which should be updated to the current SAML config (id={new_config.id})',
+            output
+        )
+        self.assertIn('CHECK SUMMARY:', output)
+        self.assertIn('Providers: 2', output)
+        self.assertIn('Outdated: 1', output)
+
+        # Check key observability calls
+        expected_calls = [
+            mock.call('saml_management_command.operation', 'run_checks'),
+            mock.call('saml_management_command.total_providers', 2),
+            mock.call('saml_management_command.outdated_count', 1),
+        ]
+        for call in expected_calls:
+            self.assertIn(call, mock_set_custom_attribute.call_args_list)
```

Comment:
@ktyagiapphelix2u: Elsewhere in this same PR you used the following. Why not here?
```suggestion
        mock_set_custom_attribute.assert_has_calls(expected_calls, any_order=False)
```